src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1468,6 +1475,92 @@ private function registerPropertyInfoConfiguration(array $config, ContainerBuild
+                            $connectionDefinition->setPublic(false);
+                            $connectionDefinition->setFactory(array(StoreFactory::class, 'createConnection'));
+                            $connectionDefinition->setArguments(array($storeDsn));
+                            $container->setDefinition($connectionDefinitionId, $connectionDefinition);
+                        }
+
+                        $storeDefinition = new Definition(\stdClass::class);
+                        $storeDefinition->setFactory(array(StoreFactory::class, 'createStore'));
+                        $storeDefinition->setArguments(array(new Reference($connectionDefinitionId)));
+
+                        $container->setDefinition($storeDefinitionId = 'lock.'.$resourceName.'.store.'.md5($storeDsn), $storeDefinition);
+
+                        $storeDefinition = new Reference($storeDefinitionId);
+                        break;
+                    case $usedEnvs:
+
+                        break;
+                    default:
+                        throw new InvalidArgumentException(sprintf('Lock store DSN "%s" is not valid in resource "%s"', $storeDsn, $resourceName));
+                }
+
+                $storeDefinitions[] = $storeDefinition;
+            }
+
+            // Wrap array of stores with CombinedStore
+            if (count($storeDefinitions) > 1) {
+                $combinedDefinition = new ChildDefinition('lock.store.combined.abstract');
+                $combinedDefinition->replaceArgument(0, $storeDefinitions);
+                $container->setDefinition('lock.'.$resourceName.'.store', $combinedDefinition);
+            } else {
+                $container->setAlias('lock.'.$resourceName.'.store', new Alias((string) $storeDefinitions[0]));
+            }
+
+            // Generate factories for each resource
+            $factoryDefinition = new ChildDefinition('lock.factory.abstract');
+            $factoryDefinition->replaceArgument(0, new Reference('lock.'.$resourceName.'.store'));
+            $factoryDefinition->setPublic(true);
+            $container->setDefinition('lock.'.$resourceName.'.factory', $factoryDefinition);
+
+            // Generate services for lock instances
+            $lockDefinition = new Definition(Lock::class);
+            $lockDefinition->setFactory(array(new Reference('lock.'.$resourceName.'.factory'), 'createLock'));
+            $lockDefinition->setArguments(array($resourceName));
+            $container->setDefinition('lock.'.$resourceName, $lockDefinition);
+
+            // provide alias for default resource
+            if ('default' === $resourceName) {
+                $container->setAlias('lock.store', new Alias('lock.'.$resourceName.'.store'));
+                $container->setAlias('lock.factory', new Alias('lock.'.$resourceName.'.factory'));
+                $container->setAlias('lock', new Alias('lock.'.$resourceName));
```

Comment:
ContainerAware objects are still supported, but I think we all agree that they should not be promoted (for end applications esp.) and features should not be designed with that in mind.
Better inject `@lock.factory` (controllers/commands/everything as services) rather than locate it using a DIC (here is the bad habit: a DIC is not meant to be used as a service locator but to wire services, see https://github.com/symfony/symfony/issues/21623 for reference).

If no internal need for `get()` this service, we should keep it private (regarding BC we can make it public afterwards if legit use case, not the inverse).